src/compiler/transformers/ts.ts

```diff
@@ -1189,7 +1189,7 @@ namespace ts {
             //
 
             const prefix = getClassMemberPrefix(node, member);
-            const memberName = getExpressionForPropertyName(member, /*generateNameForComputedPropertyName*/ true);
+            const memberName = getExpressionForPropertyName(member, /*generateNameForComputedPropertyName*/ !hasSyntacticModifier(member, ModifierFlags.Ambient));

```

Comment:
The test results look right to me, but I don't understand why non-ambient names need a generated name in the first place. Can you explain?

Reply:
@sandersn If we're talking about _`ambient`_ context and decorators, I think we can avoid the new variables and use the original expression., because _`ambient`_ elements are skipped in JavaScript files. Maybe I'm wrong and we need to use a temp variable to store _computed property name for _`ambient`_ context  too 

Do you prefer to have the output for non-ambient names like this 

```ts
function dec(klass, name) {}
const isSymbol = Symbol('isSymbol');
class Foo {
}
__decorate([
    dec,
    __metadata("design:type", Number)
], Foo.prototype, isSymbol, void 0);
```

instead of 

```ts
var _a;
function dec(klass, name) {}
const isSymbol = Symbol('isSymbol');
class Foo {
}
_a = isSymbol;
__decorate([
    dec,
    __metadata("design:type", Number)
], Foo.prototype, _a, void 0);
```

Need to check this case, maybe there are cases in which a temporary variable is required to store a member name.

Reply:
The first output looks better to me. But I'm not aware of complex emit cases the way @rbuckton is, so there might be some reason that the alias is needed.

Reply:
I think this is an acceptable change. There's no need to introduce a temp that is only used once.